pkg/provider/cloud/azure/vnet.go

```diff
@@ -130,6 +130,17 @@ func deleteVNet(ctx context.Context, clients *ClientSet, cloud kubermaticv1.Clou
 	if cloud.Azure.VNetResourceGroup != "" {
 		resourceGroup = cloud.Azure.VNetResourceGroup
 	}
+
+	// We first do Get to check existence of the VNet to see if its already gone or not.
+	// We could also directly call delete but the error response would need to be unpacked twice to get the correct error message.
+	res, err := clients.Networks.Get(ctx, resourceGroup, cloud.Azure.VNetName, "")
+	if err != nil {
+		return err
+	}
+	if isNotFound(res.Response) {
+		return nil
+	}
```

Comment:
I think this is not going to work because the `Get` method will return an error for everything that isn't HTTP status code `200`: https://github.com/Azure/azure-sdk-for-go/blob/v63.1.0/services/network/mgmt/2021-05-01/network/securitygroups.go#L264-L272

```suggestion
	res, err := clients.Networks.Get(ctx, resourceGroup, cloud.Azure.VNetName, "")
	if err != nil {
	    if isNotFound(res.Response) {
		    return nil
	    }
		return err
	}
	
```

WDYT?

Reply:
yeah, you are right ;)